umm-spec-lib/src/cmr/umm_spec/validation/data_date.clj

```diff
@@ -0,0 +1,43 @@
+(ns cmr.umm-spec.validation.data-date
+  "Defines validations for UMM collection DataDates and MetadataDates"
+  (:require
+   [clj-time.core :as time]
+   [cmr.umm-spec.date-util :as date-util]))
+
+(defn data-dates-warning-validation
+  "Validate that DataDates/MetadataDates' 
+   CREATE, UPDATE, REVIEW and DELETE date values are in the right order and range: 
+   CREATE needs to be <= UPDATE and they both need to be in the past. 
+   REVIEW needs to be <= DELETE and they both need to be in the future.
+   field-path is :DataDates/:MetadataDates, 
+   value is the value of :DataDates/:MetadataDates."
+  [field-path value]
+  (let [latest-create (date-util/latest-date-of-type value "CREATE")
+        earliest-update (date-util/earliest-date-of-type value "UPDATE")
+        latest-update (date-util/latest-date-of-type value "UPDATE")
+        earliest-review (date-util/earliest-date-of-type value "REVIEW")
+        latest-review (date-util/latest-date-of-type value "REVIEW")
+        earliest-delete (date-util/earliest-date-of-type value "DELETE")
+        warning-msgs (concat 
```

Comment:
I'm going to merge it for now. If you guys still feel strongly about breaking it up, we can talk about it and think of some good function names... What I don't like is to validate CREATE in both validate-create-and-update and validate0date-order.  To me, it's less clear than giving a good doc string and combine them together.